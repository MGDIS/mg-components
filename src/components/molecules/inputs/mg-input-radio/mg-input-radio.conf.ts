/**
 * type RadioOption
 * use to match radio attributes
 */
export type RadioOption = {
  title: string;
  value: any;
  disabled?: boolean;
};
