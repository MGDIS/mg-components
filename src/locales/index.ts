import en from './en/messages.json';
import fr from './fr/messages.json';

const defaultLocale = 'en';
const messages = { en, fr };

/**
 * Get locale
 *
 * @param {HTMLElement} element element that needs to know the locale to use
 * @returns {string} locale
 */
const getLocale = (element: HTMLElement): string => {
  return (element.closest('[lang]') as HTMLElement)?.lang || navigator.language || defaultLocale;
};

/**
 * Get Intl object
 * We load the defined locale but for now we only support the first subtag for messages
 *
 * @param {HTMLElement} element element we need to get the language
 * @returns {{ locale: string; messages: unknown }} messages object
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const initLocales = (element: HTMLElement): { locale: string; messages: any } => {
  // Get local
  const locale = getLocale(element);
  // Only keep first subtag
  const localeSubtag = locale.split('-').shift();
  // Return
  return {
    locale,
    messages: messages[localeSubtag] || messages[defaultLocale],
  };
};
