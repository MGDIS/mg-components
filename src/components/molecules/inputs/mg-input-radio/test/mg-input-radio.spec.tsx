import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgInputRadio } from '../mg-input-radio';
import { messages } from '../../../../../locales';

const getPage = (args) => newSpecPage({
  components: [MgInputRadio],
  template: () => (<mg-input-radio {...args}></mg-input-radio>)
});

describe('mg-input-radio', () => {
  test.each([
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane']},
    {label: 'label', identifier: "identifier", items: [{ title: 'batman', value: 'u' },{ title: 'robin', value: 'i' }, { title: 'jocker', value: 'o' }, { title: 'bane', value: 'a' }]},
    {label: 'label', identifier: "identifier", items: [{ title: 'batman', value: 1 },{ title: 'robin', value: 2 }, { title: 'jocker', value: 3 }, { title: 'bane', value: 4 }]},
    {label: 'label', identifier: "identifier", items: [{ title: 'batman', value: true },{ title: 'robin', value: false }, { title: 'jocker', value: false }, { title: 'bane', value: false }]},
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane'], labelOnTop: true},
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane'], labelHide: true},
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane'], inputVerticalList: true},
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane'], required: true},
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane'], readonly: true},
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane'], disabled: true},
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane'], helpText: 'Hello jocker'},
    {label: 'label', identifier: "identifier", items: ['batman', 'robin', 'jocker', 'bane'], tooltip: "My Tooltip Message"},
  ])('Should render with args %s:', async (args) => {
    const { root } = await getPage(args);
    expect(root).toMatchSnapshot();
  });

  test.each(["", undefined])('Should throw error with invalid label property : %s', async (value) => {
    try {
      await getPage({label:value, items: ['batman', 'robin', 'jocker', 'bane']});
    }
    catch (err) {
      expect(err.message).toMatch('prop "label" is required')
    }
  });

  test.each([
    [['batman', {title:'batman', value:'batman'}]],
    [['batman', {batman:'batman'}]],
    [[true, false]],
    [[1, 2, 3]],
    [[true, 1, 'batman']],
    [[{title:'batman', value:'batman'}, {batman:'batman'}]],
    [[{title:'batman', value: undefined}, {title:'batman', value: 'test'}]],
  ])('Should throw error with invalid items property : %s', async (items) => {
    try {
      await getPage({label:'Label', items});
    }
    catch (err) {
      expect(err.message).toMatch('<mg-input-radio> prop "items" is required and all items must be the same type, string or RadioOption.')
    }
  });


    test.each([
      {items: ['batman', 'robin', 'jocker', 'bane'], inputValue: 'batman'},
      {items: [{ title: 'batman', value: 'u' },{ title: 'robin', value: 'i' }, { title: 'jocker', value: 'o' }, { title: 'bane', value: 'a' }], inputValue: 'a'},
      {items: [{ title: 'batman', value: 1 },{ title: 'robin', value: 2 }, { title: 'jocker', value: 3 }, { title: 'bane', value: 4 }], inputValue: 1},
      {items: [{ title: 'batman', value: true },{ title: 'robin', value: false }], inputValue: true}
    ])('Should trigger events for items (%s) with inputValue (%s)', async ({items, inputValue})=> {
    const args = {label: 'label', items , identifier: "identifier", helpText: "My help text"};
    const page = await getPage(args);

    const element = page.doc.querySelector('mg-input-radio');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn(()=> true);
    Object.defineProperty(input, 'validity', { get: jest.fn(()=> ({
      valueMissing: false,
    }))});

    jest.spyOn(page.rootInstance.valueChange, 'emit');

    input.dispatchEvent(new CustomEvent('focus', { bubbles: true }));
    await page.waitForChanges();

    expect(page.root).toMatchSnapshot(); //Snapshot on focus

    input.value = inputValue.toString();
    input.dispatchEvent(new CustomEvent('input', { bubbles: true }));
    await page.waitForChanges();
    expect(page.rootInstance.valueChange.emit).toHaveBeenCalledWith(inputValue);
  });

  test.each([
    {validity: true, valueMissing: false},
    {validity: false, valueMissing: true},
  ])('validity (%s), valueMissing (%s)', async ({validity, valueMissing})=> {
    const args = {label: 'label', items: ['batman', 'robin', 'jocker', 'bane'], identifier: "identifier", patternErrorMessage: "Non"};
    const page = await getPage(args);

    const element = page.doc.querySelector('mg-input-radio');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn(()=> validity);
    Object.defineProperty(input, 'validity', { get: jest.fn(()=> ({
      valueMissing,
    }))});

    input.dispatchEvent(new CustomEvent('blur', { bubbles: true }));
    await page.waitForChanges();

    if(validity) {
      expect(page.rootInstance.errorMessage).toBeUndefined();
    }
    else if (valueMissing){
      expect(page.rootInstance.errorMessage).toEqual(messages.errors.required);
    }
    expect(page.rootInstance.valid).toEqual(validity);
    expect(page.rootInstance.invalid).toEqual(!validity);
  });
});
