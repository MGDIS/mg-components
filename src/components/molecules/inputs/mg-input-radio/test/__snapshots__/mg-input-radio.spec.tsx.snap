// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-input-radio Should render with args {
  label: 'label',
  identifier: 'identifier',
  items: [Array],
  disabled: true
}: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" disabled="" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" disabled="" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" disabled="" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" disabled="" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args {
  label: 'label',
  identifier: 'identifier',
  items: [Array],
  helpText: 'Hello jocker'
}: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          Hello jocker
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args {
  label: 'label',
  identifier: 'identifier',
  items: [Array],
  inputVerticalList: true
}: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio mg-input__input-group-container--vertical">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args {
  label: 'label',
  identifier: 'identifier',
  items: [Array],
  labelHide: true
}: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title class="sr-only" identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args {
  label: 'label',
  identifier: 'identifier',
  items: [Array],
  labelOnTop: true
}: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--label-on-top mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args {
  label: 'label',
  identifier: 'identifier',
  items: [Array],
  readonly: true
}: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio mg-input--readonly">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong></strong>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args {
  label: 'label',
  identifier: 'identifier',
  items: [Array],
  required: true
}: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" required="" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" required="" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" required="" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" required="" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args {
  label: 'label',
  identifier: 'identifier',
  items: [Array],
  tooltip: 'My Tooltip Message'
}: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
          <mg-tooltip identifier="identifier-tooltip" message="My Tooltip Message">
            <mg-icon icon="info"></mg-icon>
          </mg-tooltip>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args { label: 'label', identifier: 'identifier', items: [Array] }: 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args { label: 'label', identifier: 'identifier', items: [Array] }: 2`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="u">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="i">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="o">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="a">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args { label: 'label', identifier: 'identifier', items: [Array] }: 3`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="1">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="2">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="3">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="4">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should render with args { label: 'label', identifier: 'identifier', items: [Array] }: 4`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="true">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="false">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="false">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="false">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should trigger events for items ({ items: [Array], inputValue: 'a' }) with inputValue (%s) 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="is-focused mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_0" name="identifier" type="radio" value="u">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_1" name="identifier" type="radio" value="i">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_2" name="identifier" type="radio" value="o">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_3" name="identifier" type="radio" value="a">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should trigger events for items ({ items: [Array], inputValue: 'a' }) with inputValue (%s) 2`] = `
<mg-input-radio valid="" value="a">
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="a">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="i">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="o">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" checked="" id="identifier_3" name="identifier" type="radio" value="a">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should trigger events for items ({ items: [Array], inputValue: 'batman' }) with inputValue (%s) 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="is-focused mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should trigger events for items ({ items: [Array], inputValue: 'batman' }) with inputValue (%s) 2`] = `
<mg-input-radio valid="" value="batman">
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" checked="" id="identifier_0" name="identifier" type="radio" value="batman">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="robin">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="jocker">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="bane">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should trigger events for items ({ items: [Array], inputValue: 1 }) with inputValue (%s) 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="is-focused mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_0" name="identifier" type="radio" value="1">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_1" name="identifier" type="radio" value="2">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_2" name="identifier" type="radio" value="3">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_3" name="identifier" type="radio" value="4">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should trigger events for items ({ items: [Array], inputValue: 1 }) with inputValue (%s) 2`] = `
<mg-input-radio valid="" value="1">
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" checked="" id="identifier_0" name="identifier" type="radio" value="1">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="2">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_2" name="identifier" type="radio" value="3">
              <label htmlfor="identifier_2">
                jocker
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_3" name="identifier" type="radio" value="4">
              <label htmlfor="identifier_3">
                bane
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should trigger events for items ({ items: [Array], inputValue: true }) with inputValue (%s) 1`] = `
<mg-input-radio>
  <mock:shadow-root>
    <fieldset class="is-focused mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_0" name="identifier" type="radio" value="true">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="identifier-help-text" id="identifier_1" name="identifier" type="radio" value="false">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;

exports[`mg-input-radio Should trigger events for items ({ items: [Array], inputValue: true }) with inputValue (%s) 2`] = `
<mg-input-radio valid="" value="">
  <mock:shadow-root>
    <fieldset class="mg-input mg-input--radio">
      <mg-input-title identifier="identifier" is-legend="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input-container__input">
          <ul class="mg-input__input-group-container mg-input__input-group-container--radio">
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_0" name="identifier" type="radio" value="true">
              <label htmlfor="identifier_0">
                batman
              </label>
            </li>
            <li class="mg-input__input-group">
              <input aria-describedby="" id="identifier_1" name="identifier" type="radio" value="false">
              <label htmlfor="identifier_1">
                robin
              </label>
            </li>
          </ul>
        </div>
        <div class="mg-input__input-container__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </fieldset>
  </mock:shadow-root>
</mg-input-radio>
`;
